lighthouse/classes/event_properties/definitions.py

```diff
@@ -0,0 +1,213 @@
+from functools import cached_property
```

Comment:
I think this file is too long, can we move some of the classes into their own files? We need to avoid getting into a situation like https://github.com/sanger/crawler/blob/develop/crawler/file_processing.py